### PR TITLE
Hrrngh, Leader, I'm trying to sneak around, but I'm dummy thick, and the clap of my martial arts is alerting the crew.

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request
Gives CQC to traitor uplinks and removes it from clownops uplinks, and
Removes SCARP from traitor uplinks and puts it into nukeops
nukies retain cqc for stealth options or scarp for pure monk levels

## Why It's Good For The Game
IAA now has better stealth options, CQC is powerful without being super aggressively loud like SCARP, nukies can now go full unarmed without having an eshield/shieldsuit, traitors can now go solid snake mode.

## Changelog
:cl:
tweak: clown ops has to use clown martial arts now
balance: replaced traitor scarp with cqc, adds scarp to nukie uplink
/:cl:

### CODE REVIEW

**Comment by Linzolle** [comment]
I've always wanted to do a Big Boss traitor gimmick

**Comment by DeltaFire15** [comment]
On god
> Nukies
> Sleepycarp + desword

File: code/modules/uplink/uplink_items.dm [hunk 1]
```diff
@@ -597,7 +597,7 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	item = /obj/item/book/granter/martial/carp
 	cost = 17
 	surplus = 0
-	exclude_modes = list(/datum/game_mode/nuclear, /datum/game_mode/nuclear/clown_ops)
+	include_modes = list(/datum/game_mode/nuclear)
```
Comment (Trilbyspaceclone) [inline code/modules/uplink/uplink_items.dm]: Why make this nukie only? They get better melee and shields...

**Comment by Trilbyspaceclone** [comment]
Nukie + black suit + north star gloves + carp + E-shield all at once seems kinda bad

**Comment by ghost** [comment]
yeah i'm open to keeping scarp on traitors if anyone has any strong objections

sleepy carp and desword serve the same purpose so you'd either go scarp desword, grab people, and then enable desword to kill them, or you'd go scarp gloves of north star, grab, use melee damage combos.

scarp would be weaker than esword eshield but scarp blocks bullets so
and scarp forces you to fight in melee, which doesn't matter as much for someone that was going to do full melee already, but in a pinch it could get you killed
eshield and scarp don't synergize

**Comment by YakumoChen** [comment]
I'm pretty sure giving carp to nuke ops would be pretty unbalanced, especially with a desword

Martial Arts are very stealthy, I dont know where youre getting "wildly loud" from. I dont think cqc would be bery balanced on traitors either, it has some good stuns and is pretty deadly as well.

**Comment by ghost** [comment]
carp does not have stealth potential
cqc does
carp also has stuns like cqc does, but cqc also lets you sleepypen. cqc's main purpose is for stealth, while scarp's main purpose is... to not be stealthy. don't need ranged immunity if you're not getting shot.

nuke ops with eshield+cqc+north star gloves is about as strong as scarp+desword because either way you're guaranteed to die and you can't do anything about it unless you get a lucky first stunbaton hit 

edit: using desword with scarp requires more speed than other melee options. you're limited to only people that use hotkey mode and know x/z and intents. you can also use desword/cqc to similar effect but it requires even more speed/reaction speed because you have to keep desword open until you go for the grab, then turn it off and swap hands.

i mostly want to add scarp to nukies to make unarmed/melee more viable, because, as people seem to forget, a bulldog is much stronger than a desword. scarp is still weaker than any of the nukie guns with the exclusion of like the flamethrower, biosprayer, stechkin, and anti-tank pistol (obviously not in a 1v1 comparison but in killing potential). most broken part of scarp would be adrenal implants and stimpaks, but those are already overpowered as is no matter what you're using them with

**Comment by YakumoChen** [comment]
I mean, carp is stealthy, there's no inhands or any indication you have it until you start punching people, and even then it's not totally apparent unless you are watching chatlog. Carp has only a "whiff" sound when being shot at.

CQC on the other hand is not stealthy, punching people makes a loud sound effect that is unique and only associated with the martial art.

**Comment by ghost** [comment]
carp is in no way stealthy
no stealth options and it is inherently designed for being a loud antag

cqc's stealth comes from the fact that if you want to kill someone they can't call for help. you also only make noises on like 1 or 2 combos, which you don't NEED to use because you can just ko

**Comment by YakumoChen** [comment]
"Stealthy" is defined as a weapon that can kill a crewmember quietly or is able to be hidden on one's person realistically without security finding it in a search. Carp meets the latter requirement as, once you have learned it and gotten rid of the scroll, there is no way security will know it exists outside of you blocking a projectile. If you go into a fight with carp, the other party will have no way to tell you have carp until you block a projectile (given the weapon is your fists, you won't have any inhands for carp), giving you a massive element of surprise and a huge edge in a fight until the other party can adapt (guns make terrible melee weapons). 

Sleeping Carp, outside the whiff, has no indication audibly what it does, meaning someone in an adjacent room isn't going to be alerted to someone using sleeping carp, they'll just hear a fistfight in maint. Someone not watching the chatlog won't know you have carp unless projectiles come into play, so you can get kills off without alerting the crew you have carp. CQC is incapable of this feat, its attacks have a different, distinct sound that will immediately alert everyone to the fact it's a martial art style regardless of if they're watching chat or the game screen or are in line of sight.

Carp is stealthy. It's literally listed in the "stealthy weapons" category. This is a dishonest PR and a dishonest interpretation of sleeping carp. 

Nuke ops work well as a team and giving one of them an item that blocks projectiles basically forces whoever wants to fight the ops to literally run at them in melee, which is usually an idiotic idea since the other ops are going to be heavily armed with actual guns... and the melee op won't even have to worry about friendly fire, since they're immune.

You're also taking carp away from traitors, which is one of their best melee options and much more fitting as a non-op item given security can actually deal with carp users much more competently when they aren't backed up by an entire team of elite murderdudes. CQC doesn't have utility outside of a fight (you only get block for melee attacks) and security can still shoot you, making it a piss poor melee option in comparison.

**Comment by ghost** [comment]
i don't even think i should make an actual response to this because this is just clearly bait and i don't even know you well enough to know why you are but if an external force wants me to repeat myself i can

**Comment by ghost** [comment]
sike i'll add a new response instead of just repeating the same cycle of me going "but wait a second, cqc stealthy because of how it's designed and carp loud because of how it's designed" and then other person going "you're wrong go fuck yourself" despite myself just saying i didn't feel a need to respond because, you know, inconsistency is cool

the point of this pr is to encourage new strategies, "builds", general gameplay, and gimmicks for traitors and nukies. if a maintainer or headmin comes in and says "no don't add scarp to nukies make them have to use shielded hardsuit and desword instead" then ok. the entire reason this pr was made is for variety.
also iaa rounds, where iaa can actually be stealthy by using cqc to silently takedown an enemy iaa but risk getting hunted down first by someone with a powerful ranged weapon.

oh and also to add a bit to the repetitive cycle of hateboning:
just because cqc has a combo that makes a noise doesn't mean you have to use it
actually i've already said this
i just forgot that i said it because everything i say is ignored completely lmao



god why the fuck am i writing a paragraph for someone that won't even read it


because it's for other people to read

**Comment by ghost** [comment]
https://gyazo.com/a5bc8a78da70c882918cf16105a44c3c
https://gyazo.com/cf3754ec319536fb0ca67d5962272b92

thanks

**Comment by YakumoChen** [comment]
image
it's almost like I was right

**Comment by ghost** [comment]
gonna close this pr until i know if we're using tupina's uplink changes or not

**Pull request closed without merging**